packages/dd-trace/src/startup-log.js

```diff
@@ -0,0 +1,123 @@
+'use strict'
+
+const mainLogger = require('./log')
+const path = require('path')
+const { inspect } = require('util')
+const tracerVersion = require('../lib/version')
+
+let os
+try {
+  os = require('os')
+} catch (e) {
+  // in browser
+}
+
+const logger = Object.create(mainLogger)
+logger._enabled = process.env.DD_TRACE_STARTUP_LOGS !== '0'
+
+let config
+let plugins = []
+let samplingRules = []
+
+let alreadyRan = false
+
+function getIntegrations () {
+  const integrations = []
+  for (const plugin of plugins) {
+    if (plugin.versions) {
+      try {
+        const version = require(path.join(plugin.name, 'package.json')).version
+        integrations.push(`${plugin.name}@${version}`)
+      } catch (e) {
+        integrations.push(plugin.name)
+      }
+    } else {
+      integrations.push(plugin.name)
+    }
+  }
+  return Array.from(new Set(integrations))
+}
+
+function startupLog (agentError) {
```

Comment:
How would this scale to accept other errors?

Reply:
For the moment, it's the only error in startup logging that's actually relevant to us. I'm fine with putting it in an options-object argument if you think that's more future proof? I figured it didn't matter right now since there's only one.

Reply:
Yeah it doesn't matter too much, it's just that this looks a lot like the callback syntax where you would usually only have a single error, so it can read as "this first error is the only error that will be captured".

Reply:
Turns args into an object in the latest rev.